server/src/test/replay-server.ts

```diff
@@ -78,26 +101,46 @@ export async function startTestReplayServer(t: ava.TestContext):
         proxyRes.pipe(res);
 
       } else {
-        let replayBody;
-        try {
-          replayBody = await fsExtra.readFile(replayFile);
-        } catch (e) {
-          res.statusCode = 500;
-          res.statusMessage = 'No replay file: ' + replayFile;
-          res.end();
-          return;
-        }
-        res.statusCode = 200;
-        res.end(replayBody);
+        await sendReplayFile(res, replayFile);
       }
     });
   }
 
+  async function jsonHandler(
+      req: http.IncomingMessage, res: http.ServerResponse) {
+    const replayFile =
+        path.join(replayDir, (req.url || '').replace(/\//g, '#'));
```

Comment:
Took me a second -- add a comment like "replace path delimiters with arbitrary hash character so that we create a flat file structure".

Reply:
Done.